packages/UI/src/Button.tsx

```diff
@@ -1,11 +1,90 @@
+import styled from "styled-components";
+import { colors, radius } from "./styles";
+import Text from "./font";
+
+type buttonStyle = "primary" | "secondary" | "disabled" | "destructive";
+
 interface ButtonProps {
   label: string;
   onClick?: () => void;
   type?: "button" | "submit" | "reset";
+  buttonStyle: buttonStyle;
+  disabled?: boolean;
+}
+
+interface styledButtonProps {
+  $buttonStyle: buttonStyle;
 }
 
-export const Button = ({ label, onClick, type = "button" }: ButtonProps) => (
-  <button type={type} onClick={onClick}>
-    {label}
-  </button>
-);
+export const Button = ({
+  label,
+  onClick,
+  type = "button",
+  buttonStyle = "primary",
+  disabled = false,
+}: ButtonProps) => {
+  const textColor: keyof typeof colors =
+    buttonStyle === "primary"
+      ? "accent"
+      : buttonStyle === "secondary"
+        ? "primary"
+        : buttonStyle === "disabled"
+          ? "greyText"
+          : "pause";
```

Comment:
det här hade kunnat vara en funktion med en switch för att göra det lite lättare att läsa

```suggestion
const getTextColor = (variant: ButtonVariant, disabled: boolean): keyof typeof colors => {
  if (disabled) return "greyText";
  
  switch (variant) {
    case "primary": return "accent";
    case "secondary": return "primary";
    case "destructive": return "pause";
    default: return "primary";
  }
 };
```